From 2f9e41b27c089da6e7d8a5dc083e25d2ac2486c2 Mon Sep 17 00:00:00 2001
From: Prabhakar Kushwaha <prabhakar@freescale.com>
Date: Thu, 24 Dec 2015 15:33:49 +0530
Subject: [PATCH 47/74] driver: net: fsl-mc: Remove portal id hard-coding

Management Complex firmware 9.0 has fixed the issue of dprc_destroy_container
i.e. the used portal is not return to the free pool. Which was resulting in
error ethernet driver want to use this portal via either DPL or dynamically
in Linux.

So, Remove hard-coding of portal id.

Signed-off-by: Prabhakar Kushwaha <prabhakar.kushwaha@nxp.com>
---
 drivers/net/fsl-mc/mc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/fsl-mc/mc.c b/drivers/net/fsl-mc/mc.c
index a45a30d..9e042f6 100644
--- a/drivers/net/fsl-mc/mc.c
+++ b/drivers/net/fsl-mc/mc.c
@@ -821,7 +821,7 @@ static int dprc_init(void)
 		      DPRC_CFG_OPT_OBJ_CREATE_ALLOWED |
 		      DPRC_CFG_OPT_ALLOC_ALLOWED;
 	cfg.icid = DPRC_GET_ICID_FROM_POOL;
-	cfg.portal_id = 250;
+	cfg.portal_id = DPRC_GET_PORTAL_ID_FROM_POOL;
 	err = dprc_create_container(root_mc_io, MC_CMD_NO_FLAGS,
 			root_dprc_handle,
 			&cfg,
-- 
2.1.0.27.g96db324

